client/resource_group/controller/controller.go

```diff
@@ -421,7 +369,13 @@ type groupCostController struct {
 
 	// run contains the state that is updated by the main loop.
 	run struct {
-		now time.Time
+		now             time.Time
+		lastRequestTime time.Time
+
+		// requestFailed is set true when sending token bucket request.
+		// And it is set false when reciving token bucket response.
+		// This triggers a retry attempt on the next tick.
+		requestFailed bool
```

Comment:
The usage looks not very related to the "fail", `requestInProgress` makes more sence to me.